src/logging.c

```diff
@@ -697,16 +658,56 @@ void rcutils_logging_console_output_handler(
       return;
   }
 
-  fprintf(stream, "%s\n", log_str);
+  char msg_buf[1024] = "";
+  rcutils_char_array_t msg_array = {
+    .buffer = msg_buf,
+    .owns_buffer = false,
+    .buffer_length = 0u,
+    .buffer_capacity = sizeof(msg_buf),
+    .allocator = g_rcutils_logging_allocator
+  };
+
+  char output_buf[1024] = "";
+  rcutils_char_array_t output_array = {
+    .buffer = output_buf,
+    .owns_buffer = false,
+    .buffer_length = 0u,
+    .buffer_capacity = sizeof(output_buf),
+    .allocator = g_rcutils_logging_allocator
+  };
+
+  va_list args_clone;
+  va_copy(args_clone, *args);
+  status = rcutils_char_array_vsprintf(&msg_array, format, args_clone);
+  va_end(args_clone);
+
+  if (RCUTILS_RET_OK == status) {
+    status = rcutils_logging_format_message(
+      location, severity, name, timestamp, msg_array.buffer, &output_array);
+  }
```

Comment:
I think we need an else case here where we at least either set the error state or use the `SAFE_FWRITE` macro or `fprintf(stderr, ...)` (as appears to be the style in this file) to let someone know that `vsprintf` failed.